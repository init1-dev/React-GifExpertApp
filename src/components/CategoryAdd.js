import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = ({ categories, setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    };

    const handleSubmit = ( e ) => {
        e.preventDefault();

        // This could be done as helper function
        if( inputValue.trim().length > 2 && !undefined ) {
            if(categories.includes(inputValue)) {
                alert('This category already exist');
                setInputValue('');
            } else {
                setCategories( cats => [ inputValue, ...cats ] );
                setInputValue('');
            }
        }
    };

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                name="categories"
                id="categories"
                placeholder="Search Categories"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

export default CategoryAdd;

CategoryAdd.propTypes = {
    categories: PropTypes.array.isRequired,
    setCategories: PropTypes.func.isRequired
}