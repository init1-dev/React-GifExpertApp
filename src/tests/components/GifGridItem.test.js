import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas con GifGridItem.js', () => {
    const title = 'Un título cualquiera';
    const url = 'https://localhost/imagen.jpg';
    
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('Debe mostrar el <GifGridItem /> correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });
    
    test('Debe tener un párrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    });

    test('Debe tener la imagen igual al url y alt de las props', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
        
    });

    test('Debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        // const className = div.prop('className');

        expect( div.hasClass('animate__fadeIn') ).toBe( true );
        // expect( className.includes('animate__fadeIn') ).toBe( true );

    })
        
    
})