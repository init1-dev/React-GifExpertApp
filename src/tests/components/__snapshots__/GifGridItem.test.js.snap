// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGridItem.js Debe mostrar el <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título cualquiera"
    src="https://localhost/imagen.jpg"
  />
  <p>
    Un título cualquiera
  </p>
</div>
`;
